elo4/src/App.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import GlobalStyles from './styles/global';
+import CarrinhoDeCompras from './Components/CarrinhoDeCompras'
+
+import Header from './Components/header';
+import CadastroItensVenda from './Components/CadastroItensVenda';
+import Footer from './Components/Footer';
+
+import Sidenav from './Components/Sidenav/Sidenav';
+import ListagemDeItens from './Components/ListagemDeItens/ListagemDeItens.js';
+import Produto from './Components/Produto/Produto';
+
+import Grid from '@material-ui/core/Grid';
+import { withStyles } from '@material-ui/core/styles';
+
+
+const baseUrl = 'https://us-central1-future-apis.cloudfunctions.net/elo4/products';
+
+const styles = theme => ({
+  grid: {
+    padding: '0px',
+    margin: '0px'
+  },
+})
+
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      page: 'list',
+      screen: '',
+      item: ''
+    }
+  }
+  
+  componentDidMount() {
+    this.changeScreen()
+  }
+
+  handleShowProduct = async item => {
+    
+    await this.setState({
```

Comment:
O uso do `await` nesse caso não garante que o estado terá sido alterado pois o método `this.setState` **não retorna uma Promise**.

O `this.setState` é uma função assíncrona que nos disponibiliza o seu segundo parâmetro para usarmos como callback, exemplo:

```js
constructor(props){
	super(props);
	this.state = {
		nome: "future4"
	}
}


meuMetodo = () => {

	this.setState({ nome: "soter" }, () => {
    	console.log(this.state.nome); // Aqui será impresso "soter"
    })

	console.log(this.state.nome); // Aqui será impresso "future", o valor inicial que colocamos no estado
}
```

A função que passamos para o `this.setState` só será executada quando o estado tiver sido atualizado.

2 discussões sobre o tópico:
[1.](https://github.com/facebook/react/issues/2642)
[2.](https://github.com/facebook/react/issues/12482)